futurecommander_gui/main.js

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 François CADEILLAN
+ *
+ * This file is part of FutureCommander.
+ *
+ * FutureCommander is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FutureCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FutureCommander.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+console.log('MAIN');
+
+const { NodeApplication } = require('./node');
+const app = new NodeApplication();
+
+app.run({ nw });
```

Comment:
Where does `{nw}` comes from ?

Reply:
nwjs environment bring the `nw` API object dependency.

It's not a regular node or web application, it's both with nwjs as bridge between.  
So it needs nwjs runtime ( with it's proper binary launcher _and it's proper ABI_ ).

May it exists a way to get access to that API more explicitly but load some typescript definitions into webstorm did the trick to get typing autocompletion works.